src/Psalm/Type.php

```diff
@@ -787,8 +789,26 @@ public static function intersectUnionTypes(
 
             //if a type is contained by the other, the intersection is the narrowest type
             if (!$intersection_performed) {
-                $type_1_in_2 = UnionTypeComparator::isContainedBy($codebase, $type_1, $type_2);
-                $type_2_in_1 = UnionTypeComparator::isContainedBy($codebase, $type_2, $type_1);
+                $type_1_in_2 = UnionTypeComparator::isContainedBy(
```

Comment:
When I tried to use `intersectUnionTypes` with `$allow_float_int_equality = false` I got a bug.

Seems like those parameters were not passed to every calls.